src/main/java/com/microsoft/jenkins/keyvault/BaseSecretCredentials.java

```diff
@@ -21,22 +22,22 @@
     private static final long serialVersionUID = 1L;
 
     private transient SecretGetter secretGetter;
-    private final String servicePrincipalId;
+    private final String credentialId;
     private final String secretIdentifier;
 
     public BaseSecretCredentials(CredentialsScope scope,
                                  String id,
                                  String description,
-                                 String servicePrincipalId,
+                                 String credentialId,
                                  String secretIdentifier) {
         super(scope, id, description);
         this.secretGetter = SecretGetter.DEFAULT;
-        this.servicePrincipalId = servicePrincipalId;
+        this.credentialId = credentialId;
         this.secretIdentifier = secretIdentifier;
     }
 
-    public String getServicePrincipalId() {
-        return this.servicePrincipalId;
+    public String getCredentialId() {
```

Comment:
is there any backwards compatibility issues with this change? the rest looks good to me

Reply:
Make sense. Since this is a public method, I think I should keep it there.